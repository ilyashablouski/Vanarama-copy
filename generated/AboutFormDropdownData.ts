/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AboutFormDropdownData
// ====================================================

export interface AboutFormDropdownData_titles {
  __typename: "DropDownDataType";
  data: string[];
  favourites: string[];
}

export interface AboutFormDropdownData_countries {
  __typename: "DropDownDataType";
  data: string[];
  favourites: string[];
}

export interface AboutFormDropdownData_nationalities {
  __typename: "DropDownDataType";
  data: string[];
  favourites: string[];
}

export interface AboutFormDropdownData_maritalStatuses {
  __typename: "DropDownDataType";
  data: string[];
}

export interface AboutFormDropdownData_noOfDependants {
  __typename: "DropDownDataType";
  data: string[];
}

export interface AboutFormDropdownData_noOfAdultsInHousehold {
  __typename: "DropDownDataType";
  data: string[];
}

export interface AboutFormDropdownData_propertyStatuses {
  __typename: "DropDownDataType";
  data: string[];
}

export interface AboutFormDropdownData_employmentStatuses {
  __typename: "DropDownDataType";
  data: string[];
}

export interface AboutFormDropdownData {
  __typename: "DropDownType";
  titles: AboutFormDropdownData_titles;
  countries: AboutFormDropdownData_countries;
  nationalities: AboutFormDropdownData_nationalities;
  maritalStatuses: AboutFormDropdownData_maritalStatuses;
  noOfDependants: AboutFormDropdownData_noOfDependants;
  noOfAdultsInHousehold: AboutFormDropdownData_noOfAdultsInHousehold;
  propertyStatuses: AboutFormDropdownData_propertyStatuses;
  employmentStatuses: AboutFormDropdownData_employmentStatuses;
}
