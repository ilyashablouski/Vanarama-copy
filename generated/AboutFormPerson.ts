/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AboutFormPerson
// ====================================================

export interface AboutFormPerson_emailAddresses {
  __typename: "EmailAddressType";
  uuid: string;
  primary: boolean;
  value: string;
  kind: string | null;
}

export interface AboutFormPerson_telephoneNumbers {
  __typename: "TelephoneNumberType";
  uuid: string;
  kind: string | null;
  value: string;
}

export interface AboutFormPerson_companies {
  companyType: string | null;
}

export interface AboutFormPerson {
  __typename: "PersonType";
  uuid: string;
  partyUuid: string;
  title: string | null;
  firstName: string;
  lastName: string;
  emailAddresses: AboutFormPerson_emailAddresses[];
  telephoneNumbers: AboutFormPerson_telephoneNumbers[] | null;
  dateOfBirth: any | null;
  countryOfBirth: string | null;
  nationality: string | null;
  maritalStatus: string | null;
  noOfAdultsInHousehold: string | null;
  noOfDependants: string | null;
  emailConsent: boolean | null;
  profilingConsent: boolean | null;
  smsConsent: boolean | null;
  termsAndConditions: boolean | null;
  companies: AboutFormPerson_companies[] | null;
  privacyPolicy: boolean | null;
}
