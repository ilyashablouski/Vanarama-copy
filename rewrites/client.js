/* eslint-disable no-console */
const fetch = require('node-fetch');

const {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
} = require('@apollo/client');

// const inspect = require('../inspect');

const httpLink = createHttpLink({
  uri: process.env.API_URL,
  fetch,
  headers: {
    'x-api-key': process.env.API_KEY,
  },
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
  onError: ({ networkError, graphQLErrors }) => {
    console.log('graphQLErrors', graphQLErrors);
    console.log('networkError', networkError);
  },
});

module.exports = client;
