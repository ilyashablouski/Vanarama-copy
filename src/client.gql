extend type Query {
  selectedOrderUuid: ID
  selectedDerivativeId: ID
}
