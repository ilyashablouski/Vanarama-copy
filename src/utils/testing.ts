// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/prefer-default-export
export const inputChange = (element: Element, value: string) => {
  fireEvent.input(element, {
    target: { value },
  });
};
