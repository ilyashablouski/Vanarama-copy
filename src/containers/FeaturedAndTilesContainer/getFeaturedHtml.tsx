import { gql } from '@apollo/client';
import React from 'react';
import dynamic from 'next/dynamic';
import ReactMarkdown from 'react-markdown';
import getTitleTag from '../../utils/getTitleTag';
import { getFeaturedClassPartial } from '../../utils/layout';
import RouterLink from '../../components/RouterLink/RouterLink';
import { GenericPageQueryFeatured } from '../../../generated/GenericPageQueryFeatured';
import FCWithFragments from '../../utils/FCWithFragments';
import Skeleton from '../../components/Skeleton';

const Heading = dynamic(
  () => import('@vanarama/uibook/lib/components/atoms/heading'),
  {
    loading: () => <Skeleton count={1} />,
  },
);
const Image = dynamic(
  () => import('@vanarama/uibook/lib/components/atoms/image'),
  {
    loading: () => <Skeleton count={4} />,
  },
);
const Text = dynamic(
  () => import('@vanarama/uibook/lib/components/atoms/text'),
  {
    loading: () => <Skeleton count={1} />,
  },
);

interface IFeatured {
  featured: GenericPageQueryFeatured | null | undefined;
}

export const FeaturedHtml: FCWithFragments<IFeatured> = ({ featured }) => {
  const featuredClass = getFeaturedClassPartial(featured);
  return (
    <>
      {featured && (
        <section className={`row:${featuredClass}`}>
          {featured.image?.file?.url && (
            <Image
              optimisedHost={process.env.IMG_OPTIMISATION_HOST}
              src={featured.image?.file?.url}
              alt={featured.image?.file?.fileName}
            />
          )}
          <div>
            <Heading
              color="black"
              size="lead"
              tag={
                getTitleTag(
                  featured.titleTag || 'h2',
                ) as keyof JSX.IntrinsicElements
              }
            >
              {featured.title}
            </Heading>
            <ReactMarkdown
              allowDangerousHtml
              source={featured.body || ''}
              renderers={{
                link: props => {
                  const { href, children } = props;
                  return (
                    <RouterLink
                      link={{ href, label: children }}
                      classNames={{ color: 'teal' }}
                    />
                  );
                },
                heading: props => (
                  <Text {...props} size="lead" color="darker" tag="h3" />
                ),
                paragraph: props => <Text {...props} tag="p" color="darker" />,
              }}
            />
          </div>
        </section>
      )}
    </>
  );
};

FeaturedHtml.fragments = {
  featured: gql`
    fragment GenericPageQueryFeatured on Featured {
      layout
      body
      title
      titleTag
      video
      defaultHeight
      iconList {
        text
      }
      link {
        url
        text
        legacyUrl
      }
      title
      cards {
        name
        title
        image {
          title
          description
          file {
            url
            fileName
          }
        }
        body
        link {
          text
          url
          legacyUrl
        }
      }
      image {
        title
        description
        file {
          url
          fileName
        }
      }
      testimonials {
        customerName
        summary
        rating
      }
    }
  `,
};

export default FeaturedHtml;
