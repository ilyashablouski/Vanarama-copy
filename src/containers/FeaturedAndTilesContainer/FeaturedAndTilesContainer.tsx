import React, { FC, useEffect, useState } from 'react';
import dynamic from 'next/dynamic';
import ReactMarkdown from 'react-markdown';
import SchemaJSON from 'core/atoms/schema-json';
import Breadcrumb from 'core/atoms/breadcrumb-v2';
import RouterLink from '../../components/RouterLink/RouterLink';
import { IBreadcrumb } from '../../types/breadcrumbs';
import {
  getPartnerProperties,
  isPartnerSessionActive,
} from '../../utils/partnerProperties';
import { GenericPageQuery } from '../../../generated/GenericPageQuery';
import TilesContainer from '../TilesContainer/TilesContainer';
import { FeaturedHtml } from './getFeaturedHtml';
import { getSectionsData } from '../../utils/getSectionsData';
import Head from '../../components/Head/Head';
import Skeleton from '../../components/Skeleton';

const Heading = dynamic(() => import('core/atoms/heading'), {
  loading: () => <Skeleton count={1} />,
});
const Text = dynamic(() => import('core/atoms/text'), {
  loading: () => <Skeleton count={1} />,
});

interface IProps {
  data: GenericPageQuery | undefined;
  leasingOffers?: boolean;
}

const FeaturedAndTilesContainer: FC<IProps> = ({ data, leasingOffers }) => {
  const title = getSectionsData(['metaData', 'name'], data?.genericPage);
  const body = getSectionsData(['body'], data?.genericPage);

  const featured1 = getSectionsData(
    ['sections', 'featured1'],
    data?.genericPage,
  );
  const tiles = getSectionsData(['sections', 'tiles'], data?.genericPage);
  const featured2 = getSectionsData(
    ['sections', 'featured2'],
    data?.genericPage,
  );
  const featured3 = getSectionsData(
    ['sections', 'featured3'],
    data?.genericPage,
  );
  const metaData = getSectionsData(['metaData'], data?.genericPage);
  const featuredImage = getSectionsData(['featuredImage'], data?.genericPage);
  const [breadcrumbs, setBreadcrumbs] = useState([]);

  // Check if partnership session is active to set partnership as home page link
  useEffect(() => {
    const breadcrumbsItems = metaData?.breadcrumbs?.map((el: IBreadcrumb) => ({
      link: { href: el.href || '', label: el.label },
    }));
    const partnerProperties = getPartnerProperties();
    const partnershipSessionActive = isPartnerSessionActive();
    if (partnerProperties && partnershipSessionActive) {
      breadcrumbsItems[0] = {
        link: {
          href: `/partnerships/${partnerProperties?.slug?.toLowerCase()}`,
          label: 'Home',
        },
      };
    }
    setBreadcrumbs(breadcrumbsItems);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <>
      <div className="row:title">
        <Breadcrumb items={breadcrumbs} />
        <Heading size="xlarge" color="black" tag="h1">
          {title}
        </Heading>
        <ReactMarkdown
          allowDangerousHtml
          source={body || ''}
          renderers={{
            link: props => {
              const { href, children } = props;
              return (
                <RouterLink
                  link={{ href, label: children }}
                  classNames={{ color: 'teal' }}
                />
              );
            },
            heading: props => (
              <Text {...props} size="lead" color="darker" tag="h3" />
            ),
            paragraph: props => <Text {...props} tag="p" color="darker" />,
          }}
        />
      </div>
      <FeaturedHtml featured={featured1} />
      {tiles && <TilesContainer leasingOffers={leasingOffers} tiles={tiles} />}
      <FeaturedHtml featured={featured2} />
      <FeaturedHtml featured={featured3} />
      {metaData && (
        <>
          <Head metaData={metaData} featuredImage={featuredImage} />
          <SchemaJSON json={JSON.stringify(metaData.schema)} />
        </>
      )}
    </>
  );
};

export default FeaturedAndTilesContainer;
