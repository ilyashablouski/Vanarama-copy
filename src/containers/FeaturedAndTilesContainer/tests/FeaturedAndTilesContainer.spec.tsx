import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { screen, render, waitFor } from '@testing-library/react';
import FeaturedAndTilesContainer from '../FeaturedAndTilesContainer';
import { GenericPageQuery } from '../../../../generated/GenericPageQuery';

jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/',
  }),
}));

// ARRANGE
const TITLE = `Vanarama Smiles Container`;
const BODY = ``;

const SECTIONS = {
  __typename: 'Sections',
  featured1: {
    __typename: 'Featured',
    title: 'Welcome To Vanarama Smiles',
    titleTag: 'h2',
    body:
      "As a thank you, we're giving Vanarama customers exclusive access to our rewards scheme. When you lease with us you'll earn Smiles to enjoy some of the very best offers, discounts and free gifts.\n",
    image: {
      __typename: 'Image',
      title: 'gymarama',
      description: 'Vanarama smiles gym',
      file: {
        __typename: 'File',
        url:
          '//images.ctfassets.net/3xid768u5joa/6kt6I38iXsoiLAxzJDH9Ti/f3d607d7c220c9c038f434553927878d/gymarama.jpg',
        fileName: 'gymarama.jpg',
        contentType: 'image/jpeg',
      },
    },
    layout: null,
  },
  featured2: {
    __typename: 'Featured',
    title: 'Vanarama Smiles 2',
    titleTag: 'h2',
    body:
      'Due to the Covid-19 outbreak, earning and spending Smiles points on certain rewards is now temporarily on hold. We are sorry for any inconvenience this may cause. We are working hard behind the scenes to source some of the best in-home offers for you during this difficult time. To view these offers, simply visit the My Offers page on vanaramasmiles.co.uk',
    image: null,
    layout: null,
  },
  featured3: null,
  tiles: {
    __typename: 'Tiles',
    name: "What's Included?",
    tilesTitle: "What's Included?",
    titleTag: 'h2',
    tiles: [
      {
        __typename: 'Tile',
        title: 'Breakdown Cover',
        body:
          "Wherever your vehicle breaks down, our trusted partner, the RAC, will provide assistance, so you'll never be left stranded.",
        link: null,
        image: {
          __typename: 'Image',
          title: 'Fleet-Icon',
          description: 'Fleet-Icon',
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/3a609lEJL2WBF9fBG9V6N6/9bbcee8eba6f7deecfbbab92582283e2/Fleet-Icon.png',
            fileName: 'Fleet-Icon.png',
            contentType: 'image/png',
          },
        },
      },
      {
        __typename: 'Tile',
        title: 'At Home Cover',
        body:
          "If your vehicle won't start, the RAC will fix your vehicle if you break down at, or within Â¼ mile of home. Just pop the kettle on & sit tight.",
        link: null,
        image: {
          __typename: 'Image',
          title: 'free-safe-contactless-delivery',
          description: 'free-safe-contactless-delivery',
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/7gH6pHpHcgXHt14wJlfqmz/0acc5f3d654e0a2f1e2e1e584901983e/delivery.jpg',
            fileName: 'delivery.jpg',
            contentType: 'image/jpeg',
          },
        },
      },
      {
        __typename: 'Tile',
        title: 'Recovery',
        body:
          "If your vehicle can't be fixed at the roadside, the RAC will take it for repair. All part of the service!",
        link: null,
        image: {
          __typename: 'Image',
          title: 'Insurance-Icon',
          description: 'Insurance-Icon',
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/3KhuHOZWIeppLEsCTcouiY/a384211adc3f246728f3c5fd9d8cf637/Insurance-Icon.png',
            fileName: 'Insurance-Icon.png',
            contentType: 'image/png',
          },
        },
      },
      {
        __typename: 'Tile',
        title: 'Replacement Vehicle',
        body:
          "If your vehicle's off the road needing repair, we'll keep your business moving by giving you a similar-sized vehicle for up to 7 days.",
        link: null,
        image: {
          __typename: 'Image',
          title: 'Vanarama-Price-Promise-Icon',
          description: null,
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/4v2ALClaL6YnE0NYRsd8uc/7e92f2b83eb9e594ced8269152961026/Icon-PricePromise-RGB.png',
            fileName: 'Icon-PricePromise-RGB.png',
            contentType: 'image/png',
          },
        },
      },
    ],
  },
} as any;

const DATA = {
  genericPage: {
    sections: SECTIONS,
    body: BODY,
    metaData: {
      name: TITLE,
    },
  },
} as GenericPageQuery;

describe('<FeaturedAndTilesContainer />', () => {
  it('should match snapshot', async () => {
    // ACT
    const getComponent = render(
      <MockedProvider addTypename={false}>
        <FeaturedAndTilesContainer data={DATA} />
      </MockedProvider>,
    );
    // ASSERT
    await waitFor(() => {
      expect(screen.getByText(`Vanarama Smiles Container`)).toBeInTheDocument();
    });
    const tree = getComponent.baseElement;
    expect(tree).toMatchSnapshot();
  });
});
