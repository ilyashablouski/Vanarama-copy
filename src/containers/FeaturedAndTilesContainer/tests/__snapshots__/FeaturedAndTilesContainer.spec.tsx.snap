// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedAndTilesContainer /> should match snapshot 1`] = `
<body>
  <div>
    <div
      class="row:title"
    >
      <h1
        class="heading -black -xlarge"
      >
        Vanarama Smiles Container
      </h1>
    </div>
    <section
      class="row:featured-right"
    >
      <div
        class="lazyload-wrapper"
      >
        <div
          class="lazyload-placeholder"
        />
      </div>
      <div>
        <h2
          class="heading -black -lead"
        >
          Welcome To Vanarama Smiles
        </h2>
        <p
          class="text -regular -darker"
        >
          As a thank you, we're giving Vanarama customers exclusive access to our rewards scheme. When you lease with us you'll earn Smiles to enjoy some of the very best offers, discounts and free gifts.
        </p>
      </div>
    </section>
    <section
      class="row:bg-light"
    >
      <div>
        <h2
          class="heading -black -large"
        >
          What's Included?
        </h2>
        <div
          class="row:cards-2col"
          style="padding-top: 10px;"
        >
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/3a609lEJL2WBF9fBG9V6N6/9bbcee8eba6f7deecfbbab92582283e2/Fleet-Icon.png"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                Breakdown Cover
              </span>
            </div>
            <div>
              <p
                class="text -regular -dark"
              >
                Wherever your vehicle breaks down, our trusted partner, the RAC, will provide assistance, so you'll never be left stranded.
              </p>
            </div>
          </div>
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/7gH6pHpHcgXHt14wJlfqmz/0acc5f3d654e0a2f1e2e1e584901983e/delivery.jpg"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                At Home Cover
              </span>
            </div>
            <div>
              <p
                class="text -regular -dark"
              >
                If your vehicle won't start, the RAC will fix your vehicle if you break down at, or within Â¼ mile of home. Just pop the kettle on & sit tight.
              </p>
            </div>
          </div>
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/3KhuHOZWIeppLEsCTcouiY/a384211adc3f246728f3c5fd9d8cf637/Insurance-Icon.png"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                Recovery
              </span>
            </div>
            <div>
              <p
                class="text -regular -dark"
              >
                If your vehicle can't be fixed at the roadside, the RAC will take it for repair. All part of the service!
              </p>
            </div>
          </div>
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/4v2ALClaL6YnE0NYRsd8uc/7e92f2b83eb9e594ced8269152961026/Icon-PricePromise-RGB.png"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                Replacement Vehicle
              </span>
            </div>
            <div>
              <p
                class="text -regular -dark"
              >
                If your vehicle's off the road needing repair, we'll keep your business moving by giving you a similar-sized vehicle for up to 7 days.
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="row:featured-right"
    >
      <div>
        <h2
          class="heading -black -lead"
        >
          Vanarama Smiles 2
        </h2>
        <p
          class="text -regular -darker"
        >
          Due to the Covid-19 outbreak, earning and spending Smiles points on certain rewards is now temporarily on hold. We are sorry for any inconvenience this may cause. We are working hard behind the scenes to source some of the best in-home offers for you during this difficult time. To view these offers, simply visit the My Offers page on vanaramasmiles.co.uk
        </p>
      </div>
    </section>
  </div>
</body>
`;
